core/dnsserver/server.go

```diff
@@ -65,6 +65,10 @@ func NewServer(addr string, group []*Config) (*Server, error) {
 		// set the config per zone
 		s.zones[site.Zone] = site
 		// compile custom plugin for everything
+		if site.registry != nil {
+			// this config is already computed with the chain of plugin
+			continue
+		}
```

Comment:
A ServerBlocList is not duplicated when  it contains a BIND with mutiple addresses (or several BIND that lead to several addresses). 
The SAME ServerBlocList that is used from all of the socket listeners to process the DNS query.